kits/react-ts-eslint/src/index.tsx

```diff
@@ -1,7 +1,7 @@
 import { createRoot } from 'react-dom/client';
 import './style.css';
 
-const App = (): JSX.Element => {
+const App = () => {
```

Comment:
Co spíš:

```diff
-const App = () => {
+const App: FunctionComponent = () => {
```
?
Než ty typy (`JSX.Element`) mazat úplně?

Reply:
Já jsem se na to právě díval do dokumentace, jaký je aktuální oficiální postoj, a pochopil jsem to tak, že doporučené je typ neuvádět a nechat to na automatickém odvození transpilátoru/validátoru.
Přímo v dokumentaci o [TypeScriptu na react.dev](https://react.dev/learn/typescript) návratové typy funkcionálních komponent nikde nemají.
V [React TypeScript Cheatsheet](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components) také jako preferovanou variantu uvádějí neuvádět návratový typ a nechat to na odvození.
Ale pokud to tam s @podlomar chcete, klidně to tam doplním – TypeScript moc nemusím a nedovažuji se mít na tohle silný názor :-)

Reply:
Já jsem proti uvádění návratového typu, proto jsem rád za odstranění `JSX.Element`.

Reply:
Jestli se `FunctionComponent` neobjevuje v materiálech, jak zmiňuješ, tak to tam klidně nedávej. Každopádně já jsem velký příznivce zápisu `const App: FunctionComponent`. Mimo jiné kvůli tomu, že je pak jednoduché připsat props `const App: FunctionComponent<{ theme: 'dark' | 'light'} >`. Zároveň to i kontroluje, že komponenta má vždy return.